elrond-wasm-derive/src/model/endpoint.rs

```diff
@@ -33,5 +38,6 @@ pub enum PublicRole {
     CallbackRaw,
 
     /// Can only called from within the smart contract.
+    Event(EventMetadata),
```

Comment:
Please don't change the model like this. Events are not public, since they cannot be called from the outside.
The ABI generator can figure out that it is an event from the fact that is has an event auto-implementation.
Sorry, but this part will need to be reverted.